network.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Network watchdog
+#
+# Report IP address using the server API.
+# If connectivity is lost, blink warning pattern. Reboot if it isn't restored soon.
+
+hello="https://blinken.org/api/0/hello-pi"
+while [ 1 ]; do
+
+    n=0
+    curl -s "$hello" > /dev/null
+    r=`echo $?`
+    while [[ $r -ne 0 ]]; do
+        ./set_all.py 0 0 0 0
+        sleep .5
+        ./set_all.py 128 13 0 0
+	sleep .5
+
+        ((n=n+1))
+        if [[ "$n" -gt 300 ]]; then
+            echo "Network didn't come up, rebooting"
+            reboot
+        fi
+        curl -s "$hello" > /dev/null
+        r=`echo $?`
+    done
+
+    echo "Network is up"
+
+    for i in `seq 1 5`; do
+        ./set_all.py 0 0 0 0
```

Comment:
I'm  curious what happens if two scripts (udp-driver and set_all in this script) run simultaneously? I'm envisioning: /hello-pi endpoint failure, in the sleep 900 loop below, causing it to exit out of that. The check at the top of this loop immediately succeeds, and it'll echo "Network is up" and run ./set_all, all while udp-driver is running. Presumably they just clobber each other in /dev/xmas and the strand looks weird for that short duration, but does it recover? Or does it leave something desynchronized? My guess is if both write a multiple of 5 bytes it should be ok, even if interleaved.

Reply:
Empirically, they clobber each other, but I'll keep an eye on this in case there are deeper issues.